org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/validation/XbaseValidator.java

```diff
@@ -1558,6 +1558,32 @@ public void checkTypeGuardsOrder(XSwitchExpression expression) {
 			}
 		}
 	}
+	@Check
+	public void checkTypeGuardsOrderWithGenerics(XSwitchExpression expression) {
+		if (isIgnored(IssueCodes.UNREACHABLE_CASE)) {
+			return;
+		}
+		ITypeReferenceOwner owner = new StandardTypeReferenceOwner(getServices(), expression);
+		List<LightweightTypeReference> previousTypeReferences = new ArrayList<LightweightTypeReference>();
+		for (XCasePart casePart : expression.getCases()) {
+			JvmTypeReference typeGuard = casePart.getTypeGuard();
+			if (typeGuard == null) {
+				continue;
+			}
+			LightweightTypeReference typeReference = owner.toLightweightTypeReference(typeGuard);
+			LightweightTypeReference actualType = typeReference.getRawTypeReference();
+			if (actualType == null || typeReference == actualType) {
```

Comment:
Would a raw List match this case and not being added?
I.e. will the following get an error?
```
switch new java.util.ArrayList<String> {
   java.util.List<Integer>: "list of integers"
   java.util.List: "list of stuff"
   default: "something else"
}
```

Reply:
@svenefftinge  yes - it was already handled by the old code. i added another testcase for that.